EBGodunov/hydro_ebgodunov_edge_state_2D.cpp

```diff
@@ -186,23 +186,35 @@ EBGodunov::ComputeEdgeState ( Box const& bx, int ncomp,
             // If we can't compute good transverse terms, don't use any d/dt terms at all
             if (apy(i-1,j,k) > 0. && apy(i-1,j+1,k) > 0.)
             {
-                Real quxl = (apx(i,j,k)*u_mac(i,j,k) - apx(i-1,j,k)*u_mac(i-1,j,k)) * q(i-1,j,k,n);
-                stl += ( - (0.5*dtdx) * quxl
-                         - (0.5*dtdy) * (apy(i-1,j+1,k)*yzlo(i-1,j+1,k  ,n)*v_mac(i-1,j+1,k  )
-                                        -apy(i-1,j  ,k)*yzlo(i-1,j  ,k  ,n)*v_mac(i-1,j  ,k  )) ) / vfrac_arr(i-1,j,k);
-                if (fq && vfrac_arr(i-1,j,k) > 0.)
+                if (iconserv[n]) {
+                    Real quxl = (apx(i,j,k)*u_mac(i,j,k) - apx(i-1,j,k)*u_mac(i-1,j,k)) * q(i-1,j,k,n);
+                    stl += ( - (0.5*dtdx) * quxl
+                             - (0.5*dtdy) * (apy(i-1,j+1,k)*yzlo(i-1,j+1,k  ,n)*v_mac(i-1,j+1,k  )
+                                            -apy(i-1,j  ,k)*yzlo(i-1,j  ,k  ,n)*v_mac(i-1,j  ,k  )) ) / vfrac_arr(i-1,j,k);
+                } else {
+                   stl += ( - (0.25*dtdy) * (v_mac(i-1,j+1,k  )   + v_mac(i-1,j  ,k  )) * 
+                                            ( yzlo(i-1,j+1,k  ,n) -  yzlo(i-1,j  ,k  ,n) ) );
```

Comment:
@esclapez The non-conservative piece here is the same as for the non-EB case. Why isn't the presence of an EB changing this term?

Reply:
It is changing in that this time derivative term is not included if the area fractions are zeros. The term itself includes a cell-averaged umac and the cell-centered gradient of the face states:
 - the first piece is done with a arithmetic average from the faces umac, I've also tried an area-weighted average which did not show much difference. I'm not sure sure which way is better or if there is a rationale for choosing one or the other.
 - the second piece, I went this way to ensure that a uniform field remains uniform. It turns out, this is also the way it is implemented in hydro_ebgodunov_corner_couple.H, but I'm not sure if it's any indication.
 
I'm pretty sure I don't fully understand the reasons for the differences between the regular and the EB Godunov for the conservative update (outside of the area scaling part), so maybe I'm missing something ?